ddtrace/profiling/collector/_memalloc_heap.c

```diff
@@ -21,6 +27,17 @@ typedef struct
 
 static heap_tracker_t global_heap_tracker;
 
+
+static uint32_t
+heap_tracker_next_sample_size(uint32_t sample_size)
+{
+    // 0x4000000 == pow(2, 26)
+    uint32_t q = random_range(0x4000000);
+    double log_val = log2(q) - 26;
+    return (uint32_t) (log_val * (- log(2) * (sample_size + 1)));
```

Comment:
What is the need for `log(2)`?

Reply:
Well the result is totally off if you remove `log(2)` here :)